### ISSUE TITLE
Can see blocked content and replies to blocked users

### ISSUE THREAD

**Issue by georgeowell** [fraction/oasis#202]
## What's the problem you want solved?
I can currently see content from users I have blocked. I can also view their content on their profiles including blobs.

I can also see replies to threads that the people I have blocked have posted.

## Is there a solution you'd like to recommend?
- Make it impossible to see content from blocked users
- Ensure replication of blocked users isn't happening
- Hide replies to blocked users

### Nice to have
- Delete content of blocked users from your local database

cc: #93

**Comment by christianbundy** [comment]
Thanks for reporting this! It sounds like you're saying there are two bugs:

- The profile view shows content from blocked users when you go to their page.
- The thread view shows content from blocked users when you look at a thread that they reply to.

Is that right? If so:

- What info should be viewable on the profile page? Just their name (no photo, description, or posts)?

**Comment by georgeowell** [comment]
> The profile view shows content from blocked users when you go to their page.

Yep

> The thread view shows content from blocked users when you look at a thread that they reply to.

Yes. Also I see replies to threads of friends that a blocked person has written in the latest feed.

> What info should be viewable on the profile page? Just their name (no photo, description, or posts)?

Probably just their name and key and that you have blocked them.

**Comment by christianbundy** [comment]
Thanks!

> Also I see replies to threads of friends that a blocked person has written in the latest feed

Just to make sure I understand, you're saying that in the Latest feed you see messages where someone is responding to a thread where they're blocked by the author, even though you don't block them. If they're in the Latest feed, that means you follow them, so it seems like they should be there, right? Sorry, I don't think I understand this part 100% yet, could you add some more details or info about your expectations?

**Comment by georgeowell** [comment]
Lol that is a riddle you just posted. It's much simpler. 

I can see "User commented on thread" posts in the "Latest" feed that are comments from people I have blocked. I can't post a screenshot as there is no pagination of older posts yet :smile:

**Comment by christianbundy** [comment]
Oh, I understand -- you see people you blocked in the Latest feed. Hmm. The code should only allow messages from you and the people you follow, and *definitely* not show the people that you block. Could you confirm that this is happening on the latest version? If you want more messages you can edit `maxMessages` [here](https://github.com/fraction/oasis/blob/0259a949c4069047be6b26af3950552808b21d0f/src/models.js#L304) so it'll load more messages. It'll just be slower and the page will be longer, but it's fine for debugging.

**Comment by georgeowell** [comment]
Ok so I can't see any blocked content in "Latest".

But I am seeing blocked users in "Popular" so maybe that was it.

I am also seeing comments from friends on a thread that a blocked person has created in "Latest" and also replies from friends to blocked people in "Latest".

For example, if there was an argument and you blocked someone because of it. You don't want to see all your friend continuing the argument with that person.

This also kind of relates to the concept of "muting threads" in #93.

**Comment by christianbundy** [comment]
Ooh, popular -- I can imagine why that would happen!

> For example, if there was an argument and you blocked someone because of it. You don't want to see all your friend continuing the argument with that person.

Maybe could we discuss this in another issue? It'd be a big change, because we'd need to also warn people from commenting on a thread if they're blocked by the author, and it also causes weird problems where it deletes messages from previous discussions between the blocker and the blockee.

**Comment by georgeowell** [comment]
> Maybe could we discuss this in another issue? It'd be a big change, because we'd need to also warn people from commenting on a thread if they're blocked by the author, and it also causes weird problems where it deletes messages from previous discussions between the blocker and the blockee.

Yep, this is a separate issue and I believe it is better solved by a "Mute thread" type action which is not on the roadmap right now.
